File: Source/Extensions/Tessa.Extensions.Default.Client/Workflow/Wf/WfTaskSatelliteUIExtension.cs
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.Extensions.Default.Shared;
using Tessa.Platform.Storage;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Tasks;

namespace Tessa.Extensions.Default.Client.Workflow.Wf
{
    public class WfTaskSatelliteUIExtension :
        CardUIExtension
    {
        #region Private Methods

        protected void OnCardModelInitialized(object sender, CardModelInitializingEventArgs e)
        {
            StringDictionaryStorage<CardSection> sections = e.CardModel.Card.TryGetSections();
            if (sections == null)
            {
                return;
            }

            string docTypeTitle = sections.TryGetValue("WfTaskCardsVirtual", out var section)
                ? section.RawFields.TryGet<string>("DocTypeTitle")
                : null;

            if (!string.IsNullOrWhiteSpace(docTypeTitle))
            {
                e.WorkspaceInfo = docTypeTitle;
            }
        }

        #endregion

        #region Base Overrides

        public override async Task Initialized(ICardUIExtensionContext context)
        {
            ICardEditorModel editor;

            if (context.Model.InSpecialMode()
                && context.Model.Flags.HasNot(CardModelFlags.EditTemplate)
                && context.Model.Flags.HasNot(CardModelFlags.ViewExported)
                || (editor = context.UIContext.CardEditor) == null)
            {
                return;
            }

            editor.CardModelInitialized -= OnCardModelInitialized;

            if (context.Model.CardType.ID == DefaultCardTypes.WfTaskCardTypeID)
            {
                editor.CardModelInitialized += OnCardModelInitialized;

                TaskHistoryViewModel taskHistory = context.Model.TryGetTaskHistory();
                if (taskHistory != null)
                {
                    taskHistory.HideOpenViewCommand = true;
                }
            }
        }

        #endregion
    }
}

File: WebClient SDK/src/default/workflow/wf/wfTaskSatelliteUIExtension.ts
import {
  CardUIExtension,
  ICardUIExtensionContext,
  CardModelFlags,
  CardModelInitializingEventArgs
} from 'tessa/ui/cards';
import { hasNotFlag } from 'tessa/platform';

export class WfTaskSatelliteUIExtension extends CardUIExtension {
  public initialized(context: ICardUIExtensionContext) {
    if (
      (context.model.inSpecialMode &&
        hasNotFlag(context.model.flags, CardModelFlags.EditTemplate) &&
        hasNotFlag(context.model.flags, CardModelFlags.ViewExported)) ||
      !context.uiContext.cardEditor
    ) {
      return;
    }

    const editor = context.uiContext.cardEditor;
    editor.cardModelInitialized.remove(WfTaskSatelliteUIExtension.onCardModelInitialized);
    if (context.model.cardType.id === 'de75a343-8164-472d-a20e-4937819760ac') {
      editor.cardModelInitialized.add(WfTaskSatelliteUIExtension.onCardModelInitialized);

      const taskHistory = context.model.tryGetTaskHistory();
      if (taskHistory) {
        taskHistory.hideOpenViewCommand = true;
      }
    }
  }

  public static async onCardModelInitialized(e: CardModelInitializingEventArgs) {
    const sections = e.cardModel.card.tryGetSections();
    if (!sections) {
      return;
    }

    const section = sections.tryGet('WfTaskCardsVirtual');
    const docTypeTitle = !!section ? section.fields.tryGet('DocTypeTitle') : null;

    if (!docTypeTitle) {
      e.workspaceInfo = docTypeTitle;
    }
  }
}
